apiserver/charms.go

```diff
@@ -241,39 +242,39 @@ func (h *charmsHandler) processPost(r *http.Request, st *state.State) (*charm.UR
 		// There's currently no other time where it makes sense
 		// to accept repository charms through this endpoint.
 		if isImporting, err := modelIsImporting(st); err != nil {
-			return nil, errors.Trace(err)
+			return "", errors.Trace(err)
 		} else if !isImporting {
-			return nil, errors.New("charms may only be uploaded during model migration import")
+			return "", errors.New("charms may only be uploaded during model migration import")
 		}
 	}
 
 	series := query.Get("series")
 	if series != "" {
 		if err := charm.ValidateSeries(series); err != nil {
```

Comment:
This just does a regexp to see if series contains allowed characters. If the only time we use this now is during migration, I'm not sure we still need it. We will be cleaning up the upstream charm repo to get rid of all the series things.

Reply:
Yes this is the only place we call this in `juju/juju`. I'll drop